### PR TITLE
 mPwQlHID: Implementation of LANScanner (v. 0.0.1)

### PR DESCRIPTION
**Pull request by DaianAiziatov**
In addition I've also create temporary view for checking LANScanner implementation.

### CODE REVIEW

File: TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
+//
+//  TVRemoteControl-Bridging-Header.h
+//  TVRemoteControl
+//
+//  Created by Daian Aiziatov on 12/01/2019.
+//  Copyright © 2019 Daian Aiziatov. All rights reserved.
+//
```
Comment (pryaniq) [inline TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h]: Could you remove copyrights in the whole PR?

File: .swiftlint.yml [hunk 2]
```diff
@@ -67,5 +67,6 @@ nesting:
 
 excluded:
     - TVRemoteControl/Resources/R.generated.swift
+    - Carthage
```
Comment (pryaniq) [inline .swiftlint.yml:70]: I see a lot of SwiftLint warnings. Update SwiftLint and fix them, please :)

File: TVRemoteControl.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -379,8 +558,9 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_STYLE = Manual;
-				DEVELOPMENT_TEAM = "";
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = WXMC8MD6TM;
```
Comment (pryaniq) [inline TVRemoteControl.xcodeproj/project.pbxproj]: Could you revert these changes? Since we don't have a development profile with permitted devices we mustn't commit these changes into common codebase after using a personal device.

File: TVRemoteControl.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -41,10 +89,118 @@
 			isa = PBXGroup;
 			children = (
 				1922112B21AFC39F00C0A68C /* Rswift.framework */,
+				2B1AC8592220FC88009BBBF1 /* MMLanScan */,
```
Comment (pryaniq) [inline TVRemoteControl.xcodeproj/project.pbxproj]: How I can see that's not Framework. I think better to move into a separated directory in the project. For instance, `TVRemoteControl/Third-Party/MMLanScan`

File: TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+#ifndef TVRemoteControl_Bridging_Header_h
+#define TVRemoteControl_Bridging_Header_h
+
+#import "MMLANScanner.h"
+#import "LANProperties.h"
+#import "PingOperation.h"
+#import "MMLANScanner.h"
+#import "MACOperation.h"
+#import "MacFinder.h"
+#import "MMDevice.h"
```
Comment (pryaniq) [inline TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h:5]: The biggest part of these imports is not used in the project. I tried to exclude them and everything works fine. Please remove redundant imports

```
#import "MMLANScanner.h"
//#import "LANProperties.h"
//#import "PingOperation.h"
#import "MMLANScanner.h"
//#import "MACOperation.h"
//#import "MacFinder.h"
#import "MMDevice.h"
```

File: TVRemoteControl/ScannerTableViewController.swift [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    }()
+
+    private var connectedTVs = [SystemInfo]()
+
+    private lazy var scanner: LANScanner = {
+        return LANScanner()
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
```
Comment (pryaniq) [inline TVRemoteControl/ScannerTableViewController.swift:21]: Do you remember what we discussed about `MVC` approach with working with `View`? Let's arrange a workshop where we perform refactoring of this screen.

File: TVRemoteControl/ScannerTableViewController.swift [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        let frames = CGRect(x: 0,
+                            y: (self.navigationController?.navigationBar.frame.height ?? 0) + 20,
+                            width: (self.navigationController?.navigationBar.frame.width ?? 0),
+                            height: 20)
+        let progressView = UIProgressView(frame: frames)
+        progressView.translatesAutoresizingMaskIntoConstraints = false
+        return progressView
+    }()
+
+    private var connectedTVs = [SystemInfo]()
```
Comment (pryaniq) [inline TVRemoteControl/ScannerTableViewController.swift]: According to popular Swift style guides, `var connectedTVs: [SystemInfo] = []` looks more usual

[example](https://github.com/raywenderlich/swift-style-guide)

**Review by pryaniq — CHANGES_REQUESTED** [review]

File: TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h (hunk 1 above)
Comment (DaianAiziatov) [reply TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h]: Done.

File: .swiftlint.yml (hunk 2 above)
Comment (DaianAiziatov) [reply .swiftlint.yml:70]: Fixed.

File: TVRemoteControl.xcodeproj/project.pbxproj (hunk 3 above)
Comment (DaianAiziatov) [reply TVRemoteControl.xcodeproj/project.pbxproj]: Reverted.

File: TVRemoteControl.xcodeproj/project.pbxproj (hunk 4 above)
Comment (DaianAiziatov) [reply TVRemoteControl.xcodeproj/project.pbxproj]: Moved to app folder.

File: TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h (hunk 5 above)
Comment (DaianAiziatov) [reply TVRemoteControl/Supporting Files/TVRemoteControl-Bridging-Header.h:5]: Removed.

File: TVRemoteControl/ScannerTableViewController.swift (hunk 6 above)
Comment (DaianAiziatov) [reply TVRemoteControl/ScannerTableViewController.swift:21]: I am down to workshop. Agreed, that this controller is poorly designed, but it is just for testing 🐷 Although It might be reused 🤔

File: TVRemoteControl/ScannerTableViewController.swift (hunk 7 above)
Comment (DaianAiziatov) [reply TVRemoteControl/ScannerTableViewController.swift]: Linted

File: TVRemoteControl/Services/TVScanner/DataResponseError.swift [hunk 8]
```diff
@@ -0,0 +1,26 @@
+//
+//  DataResponseError.swift
+//  RemoteControlPhilips
+//
+//  Created by Daian Aiziatov on 13/01/2019.
+//  Copyright © 2019 Daian Aiziatov. All rights reserved.
+//
```
Comment (pryaniq) [inline TVRemoteControl/Services/TVScanner/DataResponseError.swift]: What about this file and 3 others? 👯‍♂️

File: TVRemoteControl.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -357,8 +673,9 @@
 			isa = XCBuildConfiguration;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_STYLE = Manual;
-				DEVELOPMENT_TEAM = "";
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = WXMC8MD6TM;
```
Comment (pryaniq) [inline TVRemoteControl.xcodeproj/project.pbxproj]: 🙈

**Review by pryaniq — CHANGES_REQUESTED** [review]

**Comment by pryaniq** [comment]
[Implement a service for scanning LAN IP addresses with validation for finding TV devices](https://trello.com/c/mPwQlHID/12-implement-a-service-for-scanning-lan-ip-addresses-with-validation-for-finding-tv-devices)

**Review by pryaniq — APPROVED** [review]
